### DISCUSSION TITLE
How can I make development in Assembly easier?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#112074] [Programming Help]
### Body

Hi,

I am going to write a firmware-based Space-Invaders remake in Assembly then if I can I'm going to blow my brains out with a BB gun, so before I do that how can I make Assembly as un-tedious and easy as possible? If I cannot blow my brains out, I'll likely just suffocate myself with a pillow or something afterword.
### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by probablynotnmp** [comment]
Hey! I know development in Assembly can be a lot so here are a few things that might help you  :)

1) Modularize: Break code into smaller functions for easier management.
2) Comments & Labels: Use descriptive comments and labels for clarity.
3) Readability: Prioritize clear code structure and variable names.
4) Macros & Templates: Simplify repetitive tasks with macros.
5) Libraries & Frameworks: Utilize existing resources for common tasks.
6) Debugging Tools: Emulators and debuggers aid in testing and debugging.
7) Incremental Development: Start with basic features and build upon them.
8) Documentation: Document code for easier understanding.
9) Practice: Finally, practice regularly to improve your proficiency.

**Reply by ghost** [reply]
I can use macros to simplify BIOS interrupts and BIOS interrupt table mapping
